### DISCUSSION TITLE
BE 챌린지 1회 기록

### DISCUSSION THREAD

**Discussion by t4e1** [3sam5oh/webtoon-recommendation-service#151] [아이디어]
# 목표 
1. AWS 주요 서비를 실무에서 어떻게 활용하는가에 대한 이해 
2. 내가 만든 백엔드 포트폴리오를 AWS 주요 인프라에 배포해본다 
3. ..그런데 이제 내가 지원할 회사에서 쓰는 서비스를 곁들인

# 목차 
1. S3 & Cloudfornt 소개 
2. 실무 활용사레 part 1
3. 내가 지원한 회사 채용 공고 돌아보기 
4. 실무 활용사례 part 2

# 목표 
1. AWS가 무엇인지 설명할 수 있다. 
2. S3의 주요 기능 및 활용 방안을 설명할 수 있다. 
3. 지원하고 싶은 회사의 지원 공고 톺아보기

# AWS 기초 
- 2006년 부터 시작했고...
- 클라우드 컴퓨팅 서비스고....
- GCP, Azure, NCP 등등 비슷한게 있고...
- 서비스가 많고.....
- AWS Cloud Practitioner Esssentials(한국어 강의) : 들으면 이해하는데 좋음

# 클라우드 컴퓨팅 
- 인터넷을 통해 IT 리소스를 제공하고 관리하는 서비스 

## 주요 특징 
1. 확장성 Scalability : 사용자가 필요에 따라 리소스를 쉽게 확장하거나 축소할 수 있음 
2. 유연성 Flexibility : 다양한 IT 환경에 맞춰 손쉽게 맞춤화 가능 
3. 비용 효율성  Cost Efficinecy : 초기 자본 투자 없이 사용한 만큼 비용 지불 
4. 가용성 Availibility : 높은 가용성과 자동 백업, 복구 기능으로 서비스 중단 최소화

- AWS 역량, Cloud 역량에 따라서 코드의 효율성이 달라질 수 있음. 

# S3 
- Simple Storage SErvice
- AWS 최초의 서비스 
- 클라우드 파일 저장소 
- AWS에서 가장 많이 사용되는 서비스 중 하나 

### 버킷 
- S3에서 데이터를 저장하는 컨테이너 
- 모든 S3 객체는 반드시 하나의 버킷에 속해 있음 
- S3 스토리지의 최상위 계층 (일종의 폴더) 
- 고유한 이름을 가져야함 (전세계에서 유일해야함) 
- 데이터 저장 위치, 권한, 버전, 수명 주기 정책 등을 관리 가능 

### 키
- 키는 S3 버킷 내에서 객체를 고유하게 식별하는 문자열 
- 키 = 경로 
- 개체를 식별하는데 사용 
- 키가 고유하다면 같은 이름의 객체는 존재할 수 있음 
- 디렉토리를 흉내낼 수 있도록 설계됨. S3내에서 폴더와 같은 역할을 수행 가능 

# Cloudfront

**Comment by jlee38266** [comment]
세상에 이것도 기록하시다니 감사합니다 👍👍👍 여기에 추가할 수 있는 부분 추가하겠습니다!
